EIPS/eip-7417.md

```diff
@@ -0,0 +1,239 @@
+---
+eip: 7417
+title: Token Converter
+description: Smart-contract service that converts token of one ERC version to another
+author: Dexaran (@Dexaran) <dexaran@ethereumclassic.org>
+discussions-to: https://ethereum-magicians.org/t/token-standard-converter/15252
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-27
+requires: 20, 165, 223
+---
+
+## Abstract
+
+There are multiple token standards on Ethereum chain currently. This EIP introduces a concept of cross-standard interoperability by creating a service that allows [ERC-20](./eip-20.md) tokens to be upgraded to [ERC-223](./eip-223.md) tokens anytime. [ERC-223](./eip-223.md) tokens can be converted back to [ERC-20](./eip-20.md) version without any restrictions to avoid any problems with backwards compatibility and allow different standards to co-exist and become interoperable and interchangeable.
+
+## Motivation
+
+The main motivation behind this EIP is to formalize the implementation of a service that performs token conversion and provide the address of the contract on mainnet in order to avoid potentially unsafe implementation from spreading.
+
+The idea is to record the contract address here after the deployment of the contract for everyone to publicly access and use.
+
+## Specification
+
```

Comment:
EIPs, generally, are for standards that require multiple parties to build compatible implementations.

From my (admittedly quick) read, it sounds like there will be exactly one Token Converter and any ERC-20 contract can upgrade using it.

So I have to ask, who are the multiple implementers that need to interoperate? Couldn't you just deploy this service, and anyone could use it? If so, I don't think this proposal needs an EIP.

Reply:
> From my (admittedly quick) read, it sounds like there will be exactly one Token Converter and any ERC-20 contract can upgrade using it.

There is precedent for this being an ERC-category EIP. I forget the EIP number, but there was one that used its EIP number repeating as the r, v, and s values to create an instance of a contract at a predetermined address across all chains.